File: src/MiniAdmin.Domain/Entities/OnlineUser.cs
namespace MiniAdmin.Domain.Entities;

public sealed class OnlineUser
{
    public Guid SessionId { get; set; }

    public Guid UserId { get; set; }

    public string UserName { get; set; } = string.Empty;

    public string RealName { get; set; } = string.Empty;

    public string? IpAddress { get; set; }

    public string? UserAgent { get; set; }

    public string? DeviceName { get; set; }

    public string? BrowserName { get; set; }

    public DateTimeOffset LoginAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset LastActiveAt { get; set; } = DateTimeOffset.UtcNow;

    public bool IsOnline { get; set; } = true;
}

File: frontend/vue-vben-admin/apps/web-antd/src/api/system/online-user.ts
import { requestClient } from '#/api/request';

export interface OnlineUserItem {
  browserName?: null | string;
  deviceName?: null | string;
  ipAddress?: null | string;
  lastActiveAt: string;
  loginAt: string;
  realName: string;
  sessionId: string;
  userAgent?: null | string;
  userId: string;
  userName: string;
}

export interface OnlineUserListParams {
  page?: number;
  pageSize?: number;
  userName?: string;
}

export interface OnlineUserListResult {
  items: OnlineUserItem[];
  total: number;
}

export async function getOnlineUserListApi(params: OnlineUserListParams) {
  return requestClient.get<OnlineUserListResult>('/system/online-user/list', {
    params,
  });
}

export async function forceLogoutOnlineUserApi(userId: string) {
  return requestClient.post<boolean>(
    `/system/online-user/${userId}/force-logout`,
  );
}

export async function forceLogoutOnlineSessionApi(sessionId: string) {
  return requestClient.post<boolean>(
    `/system/online-user/session/${sessionId}/force-logout`,
  );
}
